packages/nodes-base/nodes/HackerNews/HackerNews.node.ts

```diff
@@ -0,0 +1,176 @@
+				type: 'string',
+				required: true,
+				default: '',
+				description: 'The id of the article to be returned',
+				displayOptions: { show: { operation: [ 'article' ] } },
+			},
+			// ----------------------------------
+			//         Username
+			// ----------------------------------
+			{
+				displayName: 'Username',
+				name: 'username',
+				type: 'string',
+				required: true,
+				default: '',
+				description: 'The user to be returned',
+				displayOptions: { show: { operation: [ 'username' ] } },
+			},
+			// ----------------------------------
+			//         Query keyword
+			// ----------------------------------
+			{
+				displayName: 'Keyword',
+				name: 'keyword',
+				type: 'string',
+				default: '',
+				displayOptions: { show: { operation: [ 'query' ] } },
+				description: 'The keyword for filtering the results of the query.',
+			},
+			// ----------------------------------
+			//         Query tag
+			// ----------------------------------
+			{
+				displayName: 'Tags',
+				name: 'tags',
+				type: 'multiOptions', // undocumented
+				options: [
+					{
+						name: 'Story',
+						value: 'story',
+						description: 'Returns query results filtered by story tag',
+					},
+					{
+						name: 'Comment',
+						value: 'comment',
+						description: 'Returns query results filtered by comment tag',
+					},
+					{
+						name: 'Poll',
+						value: 'poll',
+						description: 'Returns query results filtered by poll tag',
+					},
+					{
+						name: 'Show HN',
+						value: 'show_hn', // snake case per HN tags
+						description: 'Returns query results filtered by Show HN tag',
+					},
+					{
+						name: 'Ask HN',
+						value: 'ask_hn', // snake case per HN tags
+						description: 'Returns query results filtered by Ask HN tag',
+					},
+					{
+						name: 'Front Page',
+						value: 'front_page', // snake case per HN tags
+						description: 'Returns query results filtered by Front Page tag',
+					}
+				],
+				default: '',
+				displayOptions: { show: { operation: [ 'query' ] } },
+				description: 'Tag for filtering the results of the query.',
+			},
+		]
+	};
+
+
+	async execute(this: IExecuteFunctions): Promise<INodeExecutionData[][]> {
+		const items = this.getInputData();
+		const returnData: IDataObject[] = [];
+
+		const operation = this.getNodeParameter('operation', 0) as string;
+
+		for (let i = 0; i < items.length; i++) {
+
+			let qs: IDataObject = {};
+			let endpoint = '';
+
+			if (operation === 'article') {
+				endpoint = `items/${this.getNodeParameter('id', i)}`;
+			} else if (operation === 'username') {
+				endpoint = `users/${this.getNodeParameter('username', i)}`;
+			} else if (operation === 'query') {
+				const tags = this.getNodeParameter('tags', i) as string[];
+				qs = {
+					query: this.getNodeParameter('keyword', i),
+					tags: tags ? tags.join() : ''
+				};
+				endpoint = 'search?';
+			} else {
+				throw new Error(`The operation "${operation}" is not known!`);
```

Comment:
Maybe use a try-catch block instead here

Reply:
Thanks! Can we think of a benefit of catching when it's only to rethrow? I agree with [this opinion](https://stackoverflow.com/a/42171508) in it makes sense only when you have additional work to do in the catch block.

Reply:
That's a good point! I do think that the way you wrote this code makes it very readable-- was more thinking about conventions in the codebase? But I think leaving it is definitely good!

Reply:
Agreed, it's something we should improve on in the codebase.